Trell/LocalFolderStorage.cs

```diff
@@ -18,43 +17,42 @@ public LocalFolderStorage(AbsolutePath rootPath, int maxDatabasePageCount) {
                 throw new ArgumentException("Directory must exist and could not be created.");
             }
         }
-        this.RootDirectory = d;
+        this.RootDirectory = d.FullName;
         this.MaxDatabasePageCount = maxDatabasePageCount;
     }
 
-    public bool TryResolvePath(
+    public bool TryResolveTrellPath(
         string path,
         [NotNullWhen(true)] out AbsolutePath resolvedPath,
         [NotNullWhen(false)] out TrellError? error
     ) {
         resolvedPath = default;
         error = null;
 
-        if (path == ".") {
-            resolvedPath = new AbsolutePath(this.RootDirectory.FullName);
-            return true;
+        if (Path.IsPathFullyQualified(path)) {
+            error = new TrellError(TrellErrorCode.INVALID_PATH, path);
+            return false;
         }
 
         if (!TrellPath.TryParseRelative(path, out var trellPath)) {
             error = new TrellError(TrellErrorCode.INVALID_PATH, path);
             return false;
         }
+        var relPath = string.Join(Path.DirectorySeparatorChar, trellPath.PathSegments);
+        resolvedPath = new AbsolutePath(Path.Join(this.RootDirectory, relPath));
```

Comment:
Why not `var relPath = trellPath.ToString()`? Now that TrellPath defines ToString().
Or does that cause issues on Windows since TrellPath.ToString uses / but Path.DirectorySeparatorChar on Windows is \ ?

Reply:
That would work too, and I actually had that in the code for a little bit. My reason for not going with it was that it would mean that on windows, part of the result here would be windows-style, and part of it would be linux-style (since we use GetFullPath on the root, which does that conversion if needed).

That seemed a little more likely to cause issues than being fully platform-specific, and may look goofy when debugging, but it may be fine.

Reply:
Alright, this is fine for now then. Keeping the separators consistent is useful.